Calendário Brasileiro

mileo opened the pull request:
Descrição do problema/nova funcionalidade deste Pull Resquest(PR):
------------------------------------------------------------------

Modificação do calendário do Odoo para implementar o calendário Brasileiro:
  - **Base para módulos de folha, férias e financeiro;**
  - Feriados bancários;
  - Feriados emendados;
  - Feriados Nacionais;
  - Feriados estaduais;
  - Feriados municipais
- Funções para verificação:
  - data_eh_dia_util
  - quantidade_dias_uteis em um determinado período
  - proximo_dia_util
  - data_eh_feriado_emendado
  - data_eh_feriado_bancario
- Wizard de Importação de feriados da lib pybrasil
 

Comportamento atual antes do PR:
--------------------------------
Funcionalidade inexistente

Comportamento esperado depois do PR:
------------------------------------


- [ ] Esta mudança não altera a estrutura do banco de dados, portanto não precisa de script de migração.

--
Eu confirmo que eu assinei a CLA e li as recomendações de como contribuir:
- https://odoo-community.org/page/cla
- https://odoo-community.org/page/Contribute

mileo committed:
[NEW] Module l10n_br_resource

mileo committed:
[NEW] Create l10n_br_resource module structure

mileo committed:
[NEW] Add calendar hierarchy

mileo committed:
[NEW] Add l10n_br_base fields to calendar

mileo committed:
[NEW] Add hierarchy resource calendar leaves

mileo committed:
[NEW] Melhorias nas visões e inclusao dos campos
abrangencia e tipo

hendrixcosta committed:
[FIX] Importação baseada no intervalo selecionado

hendrixcosta committed:
[FIX] Ajuste para passar nos testes

hendrixcosta committed:
[FIX] Ajuste para passar nos testes

hendrixcosta committed:
[FIX] Ajuste para passar nos testes

hendrixcosta committed:
[ADD] Add Menuitem

hendrixcosta committed:
[IMP] Função que verifica se data eh feriado

lfdivino committed:
Adicionado as funcoes de consulta data_eh_feriado_bancario() e data_eh_feriado_emendado()

lfdivino committed:
Adicionado as funcoes de consulta data_eh_feriado_bancario() e data_eh_feriado_emendado()

lfdivino committed:
Adicionado as funcoes de consulta data_eh_feriado_bancario() e data_eh_feriado_emendado()

lfdivino committed:
Adicionado funcoes para verificar dia util. Correcao nas dependencias de repositorios oca para o travis

lfdivino committed:
Adicionado testes para verificar se data eh feriado

hendrixcosta committed:
[FIX] Função que verifica se a data é feriado

hendrixcosta committed:
[ADD] Funcao que retorna proximo dia util e teste da função

hendrixcosta committed:
Flake8

[FIX] Flake8 e ajuste nos testes

hendrixcosta committed:
[FIX] Refatorando os módulos

hendrixcosta committed:
[ADD] Testes para calculo de desconto de DSR

hendrixcosta committed:
[FIX] Importação baseada no intervalo selecionado

hendrixcosta committed:
[IMP] Função que verifica se data eh feriado

hendrixcosta committed:
[FIX] Regras de acesso para passar nos testes + Flake8

hendrixcosta committed:
[FIX] Remover comentarios e arquivos desnecessarios

hendrixcosta committed:
[ADD] Add translation pt_BR

hendrixcosta committed:
[FIX] Renomeando funcoes

hendrixcosta committed:
[ADD] Teste para dias uteis

hendrixcosta committed:
[ADD] Create Module l10n_br_hr_holiday

hendrixcosta committed:
[FIX] Importação dos feriados com timezone de brasilia

hendrixcosta committed:
[REM] Remover classe inutilizada

hendrixcosta committed:
[ADD] Calculo dias uteis em periodo + test

hendrixcosta committed:
[FIX] Verificacao feriado_bancario + test

hendrixcosta committed:
[FIX] Rename funções e campos

lfdivino committed:
[FIX] no xml da rubrica de va/vr e na funcao que verifica o valor das rubricas especificas do contrato do funcionario

mileo committed:
[MIG] to v10 [REF] Removing duplicates

leorochael commented on l10n_br_resource/README.rst:
```diff
@@ -0,0 +1,80 @@
+L10n Br Resource
+================
+
+This module extend core resource to create important brazilian informations. Define a Brazilian calendar and some tools to compute dates used in financial and payroll modules
```
> This module extend core resource to create important brazilian informations.

Além de vários erros gramaticais em inglês, essa linha não faz sentido, e parece redundante com a linha de baixo.

leorochael commented on l10n_br_resource/README.rst:
```diff
@@ -0,0 +1,80 @@
+Contributors
+------------
+
+* Firstname Lastname <email.address@example.org>
```
Missing contributors. And there are many copy+paste parts from the original template that need to be filled up above.

leorochael commented on l10n_br_resource/__manifest__.py:
```diff
@@ -0,0 +1,33 @@
+{
+    'name': 'L10n Br Resource',
+    'summary': """
+        This module extend core resource to create important brazilian
```
Same comment about grammar and meaning of this part as in the README.

leorochael commented on l10n_br_resource/data/res_country_data.xml:
```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
```
What is this file used for? Seems out of place in a module about the Brazilian calendar.

leorochael commented on l10n_br_resource/demo/resource_calendar.xml:
```diff
@@ -0,0 +1,15 @@
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<openerp>
+<data noupdate="1">
```
File is empty. If you want to do add demo objects in the future, but don't have one to add now, please put this info in a TODO.rst file instead of adding useless files to the source.

And please avoid `noupdate` if possible. Makes updating objects hard.

mileo committed:
[NEW] Add pybrasil to travis.yml

[FIX] Set travis pillow==3.4.2

[NEW] Requirements file

[REM] --allow-external PyXMLSec --allow-insecure PyXMLSec

fixup! [NEW] Requirements file

leorochael commented on l10n_br_resource/i18n/pt_BR.po:
```diff
@@ -0,0 +1,232 @@
+#. module: l10n_br_resource
+#: view:wizard.workalendar.holiday.import:l10n_br_resource.view_workalendar_holiday_import
+msgid "Cancel"
+msgstr "Cancel"
```
Traduzir?

leorochael commented on l10n_br_resource/models/resource_calendar.py:
```diff
@@ -0,0 +1,263 @@
+    )
+    leave_type = fields.Selection(
+        string=u'Tipo',
+        selection=[item for item in TIPO_FERIADO.iteritems()],
```
> `selection=[item for item in TIPO_FERIADO.iteritems()],`

Can be also written as:

    selection=list(TIPO_FERIADO.iteritems()),

Or simply (in Python 2 at least):

    selection=TIPO_FERIADO.items(),

leorochael commented on l10n_br_resource/models/resource_calendar.py:
```diff
@@ -0,0 +1,263 @@
+        readonly=True
+    )
+    leave_type = fields.Selection(
+        string=u'Tipo',
```
Não deveria ser `u'Tipo de Feriado'`? Ou ainda, `Leave Kind`?

leorochael commented on l10n_br_resource/models/resource_calendar.py:
```diff
@@ -0,0 +1,263 @@
+        string=u'Tipo',
+        selection=[item for item in TIPO_FERIADO.iteritems()],
+    )
+    abrangencia = fields.Selection(
```
talvez usar `leave_scope` como nome do campo?

Em todo caso

        string=u'Abrangencia',

deveria ser

        string=u'Abrangência',

leorochael commented on l10n_br_resource/wizard/workalendar_holiday_import.py:
```diff
@@ -0,0 +1,191 @@
+try:
+    from pybrasil import feriado
+except ImportError:
+    _logger.info('Cannot import pybrasil')
```
Pelo menos `_logger.warning()` mas prefiro `_logger.error()`

leorochael reviewed (changes requested):
A quick review over the source code, reporting the more obvious things that jumped out at me.

Did not try running yet.

leorochael commented on .travis.yml:
```diff
@@ -29,6 +29,7 @@ install:
   - pip install unidecode
   - pip install suds
   - pip install http://xmlsoft.org/sources/python/libxml2-python-2.6.21.tar.gz
+  - pip install https://github.com/odoo-brazil/pybrasil/archive/master.zip
```
Please put all these in a `requirements.txt` file. No reason to invoke `pip` multiple times here.

mileo committed:
[FIX] Estados Brasileiros com preenchimento do IBGE

mileo committed:
[REF] Importação automatica do calendário durante a instalação do módulo

mileo committed:
[FIX] Translations, logger, typo

[REM] unused files

fixup! [FIX] Translations, logger, typo

mileo commented:
@leorochael 

Obrigado por revisar tão prontamente. 

Seus questionamentos foram corrigidos
